pkg/api/server/v1alpha2/plugin/plugin_logs.go

```diff
@@ -458,6 +464,241 @@ func mergeLogParts(logPath []string, re *regexp.Regexp) [][]string {
+		return status.Error(codes.Internal, "Error streaming log")
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusCreated {
+		s.logger.Errorf("Splunk Job Creation API request failed with HTTP status code: %d", resp.StatusCode)
+		return status.Error(codes.Internal, "Error fetching log data - search job creation failed")
+	}
+
+	data, err := io.ReadAll(resp.Body)
+	if err != nil {
+		s.logger.Errorf("failed to read response body, err: %s", err.Error())
+		return status.Error(codes.Internal, "Error streaming log")
+	}
+
+	var searchResponse struct {
+		SID string `json:"sid"`
+	}
+
+	if err := json.Unmarshal(data, &searchResponse); err != nil {
+		s.logger.Errorf("failed to unmarshal Splunk Search response, err: %s, data: %s", err.Error(), string(data))
+		return status.Error(codes.Internal, fmt.Sprintf("Error processing fetched log data, err: %s, data: %s",
+			err.Error(), string(data)))
+	}
+
+	if err := pollSplunkJobStatus(s, URL.String()+"/"+searchResponse.SID+outputFormat, token); err != nil {
+		s.logger.Errorf("failed to poll splunk job status, err: %v", err)
+		return status.Error(codes.Internal, fmt.Sprintf("Error failed to poll splunk search job status, err: %s", err.Error()))
+	}
+
+	req, err = http.NewRequest("GET", URL.String()+"/"+searchResponse.SID+"/results?output_mode=json_rows&count=0", nil)
+	if err != nil {
+		s.logger.Errorf("new request to splunk failed, err: %s:", err.Error())
+		return err
+	}
+
+	req.Header.Set("Authorization", "Bearer "+token)
+	lresp, err := s.client.Do(req)
+	if err != nil {
+		s.logger.Errorf("request to fetch log from  splunk failed, err: %s, req: %v", err.Error(), req)
+		return status.Error(codes.Internal, "Error streaming log")
+	}
+
+	if lresp == nil {
+		s.logger.Errorf("request to splunk failed, received nil response")
+		s.logger.Debugf("splunk request url:%s", URL.String())
+		return status.Error(codes.Internal, "Error streaming log")
+	}
+	defer lresp.Body.Close()
+
+	if lresp.StatusCode != http.StatusOK {
+		s.logger.Errorf("Splunk Fetch Log API request failed with HTTP status code: %d", resp.StatusCode)
+		return status.Error(codes.Internal, "Error fetching log data - fetch log api failed")
+	}
+
+	data, err = io.ReadAll(lresp.Body)
+	if err != nil {
+		s.logger.Errorf("failed to read response body, err: %s", err.Error())
+		return status.Error(codes.Internal, "Error streaming log")
+	}
+
+	var logData struct {
+		Rows [][]string `json:"rows"`
+	}
+	if err := json.Unmarshal(data, &logData); err != nil {
+		s.logger.Errorf("failed to unmarshal Splunk log data, err: %s, data: %s", err.Error(), string(data))
+		return fmt.Errorf("failed to unmarshal Splunk log data, err: %s", err.Error())
+	}
+
+	var logMessages []string
+	for _, msg := range logData.Rows {
+		if len(msg) != 2 {
+			s.logger.Errorf("mismatch in column data, should be 2 received: %v, %v", len(msg), msg)
+			continue
+		}
+		logMessages = append(logMessages, msg[1]+" : "+msg[0])
```

Comment:
If I understand correctly from test file, we log the step name for each line. This waste of bandwidth we return to the client and also waste of screen space for the user. I suggest we log the step once at the top, then only the log lines, like:
```
step1:
line1
line2
step2:
line1
line2
```
This is also similar to what we return for S3 logs. You can do that by keeping the name in a variable and print a new line with the step when it changes.
Obviously I assume the log lines from the different steps come in order and not as a mix, which will not make sense for the user as well.

Reply:
Yes, I have changed it to what we do for s3 logs. Splunk return logs in order.